src/phoenix/experimental/evals/functions/classify.py

```diff
@@ -139,43 +138,29 @@ def llm_classify(
     if generation_info := model.verbose_generation_info():
         printif(verbose, generation_info)
 
-    def _process_response(response: str) -> Tuple[str, Optional[str]]:
-        if not use_openai_function_call:
-            if provide_explanation:
-                unrailed_label, explanation = (
-                    eval_template.extract_label_from_explanation(response),
-                    response,
-                )
-                printif(
-                    verbose and unrailed_label == NOT_PARSABLE,
-                    f"- Could not parse {repr(response)}",
-                )
-            else:
-                unrailed_label = response
-                explanation = None
-        else:
-            try:
-                function_arguments = json.loads(response, strict=False)
-                unrailed_label = function_arguments.get(_RESPONSE)
-                explanation = function_arguments.get(_EXPLANATION)
-            except json.JSONDecodeError:
-                unrailed_label = response
-                explanation = None
-        return _snap_to_rail(unrailed_label, rails, verbose=verbose), explanation
-
     async def _run_llm_classification_async(prompt: str) -> Tuple[str, Optional[str]]:
         with set_verbosity(model, verbose) as verbose_model:
-            response = await verbose_model._async_generate(
+            unparsed_output = await verbose_model._async_generate(
                 prompt, instruction=system_instruction, **model_kwargs
             )
-        return _process_response(response)
+        return eval_template.parse_output(
```

Comment:
There's a PR out right now that tweaks this output to optionally include the prompt and unparsed response in the output dataframe and I think we might be refining this logic quite a bit over time. For this reason I somewhat prefer putting the definition of the Eval templates close to where we use them (e.g. in this file vs. `templates`, does this make sense in context of the other templates we have?